listening-platform/src/main/kotlin/com/nikitahohulia/listeningplatform/repository/CustomPostRepositoryImpl.kt

```diff
@@ -3,64 +3,68 @@ package com.nikitahohulia.listeningplatform.repository
 import com.nikitahohulia.listeningplatform.entity.Post
 import org.bson.types.ObjectId
 import org.springframework.data.domain.Sort
-import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate
 import org.springframework.data.mongodb.core.query.Criteria
 import org.springframework.data.mongodb.core.query.Query
 import org.springframework.stereotype.Repository
+import reactor.core.publisher.Flux
+import reactor.core.publisher.Mono
 
 @Repository
-class CustomPostRepositoryImpl(private val mongoTemplate: MongoTemplate) : CustomPostRepository {
+class CustomPostRepositoryImpl(private val mongoTemplate: ReactiveMongoTemplate) : CustomPostRepository {
 
-    val collectionName = "post"
-
-    override fun findPostById(id: ObjectId): Post? {
+    override fun findPostById(id: ObjectId): Mono<Post> {
         val query = Query().addCriteria(Criteria.where("id").`is`(id))
-        return mongoTemplate.findOne(query, Post::class.java, collectionName)
+        return mongoTemplate.findOne(query, Post::class.java, Post.COLLECTION_NAME)
     }
 
-    override fun findAll(): List<Post> {
-        return mongoTemplate.findAll(Post::class.java, collectionName)
+    override fun findAll(): Flux<Post> {
+        return mongoTemplate.findAll(Post::class.java, Post.COLLECTION_NAME)
     }
 
-    override fun findAllByCreatorId(id: ObjectId): List<Post> {
+    override fun findAllByCreatorId(id: ObjectId): Flux<Post> {
         val query = Query().addCriteria(Criteria.where("creatorId").`is`(id))
-        return mongoTemplate.find(query, Post::class.java, collectionName)
+        return mongoTemplate.find(query, Post::class.java, Post.COLLECTION_NAME)
```

Comment:
```suggestion
        return mongoTemplate.find(query, Post::class.java)
```

If you declare the collection name inside the annotation `@Document` then passing the collection name to mongoTemplate is not required

Reply:
```suggestion
        return mongoTemplate.find<Post>(query)
```
Also, your code could be simplified